spec/jobs/clean_up_user_api_keys_max_life_spec.rb

```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
+
+RSpec.describe Jobs::CleanUpUserApiKeysMaxLife do
+  fab!(:older_key) { Fabricate(:readonly_user_api_key, created_at: 3.days.ago) }
+  fab!(:newer_key) { Fabricate(:readonly_user_api_key, created_at: 1.day.ago) }
+
+  context "when user api key was created before the max life period" do
+    before { SiteSetting.revoke_user_api_keys_maxlife_days = 2 }
+
+    it "should revoke the key" do
+      freeze_time
+
+      described_class.new.execute({})
+
+      expect(older_key.reload.revoked_at).to eq_time(Time.zone.now)
+      expect(newer_key.reload.revoked_at).to eq(nil)
+    end
```

Comment:
For this kind of spec I think it's interesting to use the `change to` form:

```ruby
expect {
  described_class.new.execute({})
}.to change { older_key.revoked_at }.from(nil).to(eq_time(Time.zone.now))
 .and not_change { newer_key.revoked_at }
```

Didn't try it so might be wrong on syntax, but you get the idea

Reply:
Thanks @jjaffeux! I've updated the tests with this approach and added the `active` filter to the existing user API clean up job to avoid updating `revoked_at` unnecessarily